src/OrbitVulkanLayer/SubmissionTracker.h

```diff
@@ -98,6 +104,12 @@ class SubmissionTracker : public VulkanLayerProducer::CaptureStatusListener {
     std::string label_name;
     Color color;
     size_t depth;
+
+    // These are set when the timestamps were queried successfully. When these have values, the
+    // corresponding slot indices for timestamp queries have been reset. Do not query again in
+    // that case and only get the timestamps through these members.
+    std::optional<uint64_t> begin_timestamp;
```

Comment:
Nit: I am wondering if we should put the optional timestamp into the `SubmittedMarker`. Note that the state begin_info == nullopt, begin_timestamp == value would be invalid.

Reply:
Yes, I think that would make sense.

Reply:
Done.